ci/eval/compare/default.nix

```diff
@@ -0,0 +1,52 @@
+{
+  lib,
+  jq,
+  runCommand,
+  writeText,
+  supportedSystems,
+  ...
+}:
+{ beforeResultDir, afterResultDir }:
+let
+  inherit (import ./utils.nix { inherit lib; })
+    diff
+    groupByKernel
+    extractPackageNames
+    getLabels
+    ;
+
+  getAttrs = dir: builtins.fromJSON (builtins.readFile "${dir}/outpaths.json");
+  beforeAttrs = getAttrs beforeResultDir;
+  afterAttrs = getAttrs afterResultDir;
+
+  diffAttrs = diff beforeAttrs afterAttrs;
+
+  changed-paths =
+    let
+      rebuilds = lib.unique (diffAttrs.added ++ diffAttrs.changed);
```

Comment:
Some trick with unique can be applied here.

Reply:
I used https://github.com/NixOS/nixpkgs/pull/355616/files#diff-747a1e19dd64dc3c078800ad4bbcc30f4368af5c562960ac2437049684b694dbR1831 and it seems to work fine.